File: Backend-node/src/models/facialRecognition.js
const { DataTypes } = require("sequelize");
const { db } = require("../config/db");
const User = require("./user");
const FacialRecognition = db.define(
  "FacialRecognition",
  {
    id_facial_recognition: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
    },

    facial_image_url: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    status: {
      type: DataTypes.BOOLEAN,
      allowNull: false,
    },
    id_user: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: User,
        key: "id_user",
      },
    },
  },
  {
    tableName: "facial_recognition",
    timestamps: false,
  }
);

FacialRecognition.belongsTo(User, { foreignKey: "id_user" });

module.exports = FacialRecognition;

File: Backend-python/src/models/facialRecognition.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey
from sqlalchemy.orm import relationship
from config.db import Base
from models.user import User

class FacialRecognition(Base):
    __tablename__ = 'facial_recognition'

    id_facial_recognition = Column(Integer, primary_key=True, autoincrement=True)
    facial_image_url = Column(String(255), nullable=False)
    status = Column(Boolean, nullable=False)
    id_user = Column(Integer, ForeignKey('users.id_user'), nullable=False)

    user = relationship('User', back_populates='facial_recognition')

    def to_dict(self):
        return {
            'id_facial_recognition': self.id_facial_recognition,
            'facial_image_url': self.facial_image_url,
            'status': self.status,
            'id_user': self.id_user,
            'user': self.user.to_dict() if self.user else None
        }

User.facial_recognition = relationship('FacialRecognition', back_populates='user')
